seatunnel-engine/seatunnel-engine-server/src/main/java/org/apache/seatunnel/engine/server/TaskExecutionService.java

```diff
@@ -276,30 +271,21 @@ public PassiveCompletableFuture<TaskExecutionState> deployTask(
                                     ? taskGroup.getTaskGroupLocation().toString()
                                     : "taskGroupLocation is null",
                             ExceptionUtils.getMessage(t)));
-            resultFuture.complete(
-                    new TaskExecutionState(
-                            taskGroup != null && taskGroup.getTaskGroupLocation() != null
-                                    ? taskGroup.getTaskGroupLocation()
-                                    : null,
-                            ExecutionState.FAILED,
-                            t));
+            return TaskDeployState.failed(t);
         }
-        return new PassiveCompletableFuture<>(resultFuture);
     }
 
     @Deprecated
     public PassiveCompletableFuture<TaskExecutionState> deployLocalTask(
             @NonNull TaskGroup taskGroup,
             @NonNull CompletableFuture<TaskExecutionState> resultFuture) {
```

Comment:
Remove this, and modify the relevant test method

Reply:
Done